pkgs/games/funkin/default.nix

```diff
@@ -0,0 +1,105 @@
+{ stdenv, lib, fetchFromGitHub, fetchpatch
+, haxe, haxePackages, neko, makeWrapper
+, alsaLib, libpulseaudio, libGL, libX11, libXdmcp, libXext, libXi, libXinerama, libXrandr
+}:
+
+stdenv.mkDerivation rec {
+  pname = "funkin";
+  version = "unstable-2021-04-02";
+
+  src = fetchFromGitHub {
+    owner = "ninjamuffin99";
+    repo = "Funkin";
+    rev = "faaf064e37d5a150d8aba451d740eeb81bd2e974";
+    sha256 = "053qbmib9nsgx63d0rcykky6284cixibrwq9rv1sqx2ghsdn78ff";
+  };
+
+  patches = [
+    # TODO remove when fix pushed
+    # Fixes Freeplay crash on master
+    (fetchpatch {
+      url = "https://github.com/ninjamuffin99/Funkin/pull/412/commits/af62bdca4d70e06ea8b383ab5c9c6b3f5d8f087b.patch";
+      sha256 = "1p9vzm7hkay8fzrd82hgrnafd0g5qi34brzv0hyjirryhni840fy";
+    })
+  ];
+
+  postPatch = ''
+    # Real API keys are stripped from repo
+    cat >source/APIStuff.hx <<EOF
+    package;
+
+    class APIStuff
+    {
+      public static var API:String = "";
+      public static var EncKey:String = "";
+    }
+    EOF
+  '';
+
+  nativeBuildInputs = [ haxe neko makeWrapper ]
+  ++ (with haxePackages; [
+    hxcpp
+    hscript
+    openfl
+    lime
+    flixel
+    flixel-addons
+    flixel-ui
+    newgrounds
+    polymod
+    discord_rpc
+  ]);
+
+  buildInputs = [ alsaLib libpulseaudio libGL libX11 libXdmcp libXext libXi libXinerama libXrandr ];
+
+  enableParallelBuilding = true;
+
+  buildPhase = ''
+    runHook preBuild
+
+    export HOME=$PWD
+    export HXCPP_COMPILE_THREADS=${if enableParallelBuilding then "$NIX_BUILD_CORES" else "1"}
```

Comment:
Getting this propagated through the `hxcpp` package would be great. Without this the compile process uses as many threads as your CPU has in total, which is very unfriendly for obvious reasons, and it took me a good while to find this environment variable.

Reply:
This should be pretty easy to do by adding a `setupHook` to the haxe* derivations.

Reply:
Err, hxcpp of course